anndata/_core/merge.py

```diff
@@ -107,6 +109,23 @@ def equal_dataframe(a, b) -> bool:
     return a.equals(b)
 
 
+@equal.register(DaskArray)
+def equal_dask_array(a, b) -> bool:
+    import dask.array as da
+    from dask.base import tokenize
+
+    if a is b:
+        return True
+    if a.shape != b.shape:
+        return False
+    if isinstance(b, DaskArray):
+        if tokenize(a) == tokenize(b):
+            return True
+    # TODO: If we are going to give a performance warning what should it
+    # we say or recommend in the message?
+    return da.equal(a, b, where=~(da.isnan(a) == da.isnan(b))).all()
```

Comment:
I did it similarly to xarray. If I need to write a warning, what should the content be like, I am not sure. I am also not sure in which cases.

Also, one more question, I don't think we ever had a case where a and b were different types. Should I write tests for that, or do we not expect it?

Reply:
Found the code in xarray that does this: https://github.com/pydata/xarray/blob/main/xarray/core/merge.py#L149

I believe this would work for in memory and dask backed arrays.

Also it looks like we are comparing sparse arrays as equal to dense arrays, so type shouldn't matter.

```python
from scipy import sparse
import numpy as np, anndata as ad

varm_array = sparse.random(5, 200, density=0.5, format="csr")

ad1 = ad.AnnData(X=np.ones((5, 5)), varm={"a": varm_array}, dtype=np.float64)
ad2 = ad.AnnData(X=np.zeros((5, 5)), varm={"a": varm_array.A}, dtype=np.float64)

ad.concat([ad1, ad2], merge="unique")
# AnnData object with n_obs × n_vars = 10 × 5
#     varm: 'a'
```

Reply:
I wrote the tests, and they work, but I didn't understand why you send that line in the code. Should I implement the same error behavior in our case? If that is the case, it might take more time than I thought, since I would have to read deeper about the merge logic in our code.